src/complete.cpp

```diff
@@ -224,13 +224,13 @@ completion_t &completion_t::operator=(const completion_t &him) {
 }
 
 bool completion_t::is_naturally_less_than(const completion_t &a, const completion_t &b) {
+    //for this to work, stable_sort must be used because results aren't interchangeable
+    if ((a.flags & COMPLETE_DONT_SORT) != 0 && (b.flags & COMPLETE_DONT_SORT) != 0) {
```

Comment:
Maybe I misunderstood, but are we not sorting all the results for a completion and not just the results from a single completion source, and so different completion sources could have called `complete` for the same command with and without the `-k` parameter? 

I wasn't sure about the coding style in fish so I keep the explicit `!= 0` to err on the side of caution.